nodejs/aws-infra/experimental/taskDefinition.ts

```diff
@@ -0,0 +1,390 @@
+        const env1 = innerContainers[containerName] || [];
+        const env2 = options.environment || [];
+
+        const env = [...env1, ...env2];
+
+        const assignPublicIp = isFargate && !usePrivateSubnets;
+
+        // Run the task
+        const res = await ecs.runTask({
+            cluster: clusterArn,
+            taskDefinition: taskDefArn.get(),
+            placementConstraints: placementConstraints(isFargate, options.os),
+            launchType: isFargate ? "FARGATE" : "EC2",
+            networkConfiguration: {
+                awsvpcConfiguration: {
+                    assignPublicIp: assignPublicIp ? "ENABLED" : "DISABLED",
+                    securityGroups: [ securityGroupId ],
+                    subnets: subnetIds,
+                },
+            },
+            overrides: {
+                containerOverrides: [
+                    {
+                        name: "container",
+                        environment: env,
+                    },
+                ],
+            },
+        }).promise();
+
+        if (res.failures && res.failures.length > 0) {
+            console.log("Failed to start task:" + JSON.stringify(res.failures));
+            throw new Error("Failed to start task:" + JSON.stringify(res.failures));
+        }
+
+        return;
+
+        // Local functions
+        function addEnvironmentVariables(e: Record<string, string> | undefined) {
+            if (e) {
+                for (const key of Object.keys(e)) {
+                    const envVal = e[key];
+                    if (envVal) {
+                        env.push({ name: key, value: envVal });
+                    }
+                }
+            }
+        }
+    };
+}
+
+function placementConstraints(isFargate: boolean, os: HostOperatingSystem | undefined) {
+    if (isFargate) {
+        return undefined;
+    }
+
+    os = os || "linux";
+
+    return [{
+        type: "memberOf",
+        expression: `attribute:ecs.os-type == ${os}`,
+    }];
+}
+
+function computeContainerDefinitions(
+    parent: pulumi.Resource,
+    name: string,
+    containers: Record<string, mod.ContainerDefinition>,
+    logGroup: aws.cloudwatch.LogGroup): pulumi.Output<aws.ecs.ContainerDefinition[]> {
+
+    const result: pulumi.Output<aws.ecs.ContainerDefinition>[] = [];
+
+    for (const containerName of Object.keys(containers)) {
+        const container = containers[containerName];
+
+        result.push(mod.computeContainerDefinition(parent, name, containerName, container, logGroup));
+    }
+
+    return pulumi.all(result);
+}
+
+const defaultComputePolicies = [
+    aws.iam.AWSLambdaFullAccess,                 // Provides wide access to "serverless" services (Dynamo, S3, etc.)
+    aws.iam.AmazonEC2ContainerServiceFullAccess, // Required for lambda compute to be able to run Tasks
+];
+
+export function defaultTaskDefinitionTaskRolePolicies() {
+    return defaultComputePolicies.slice();
+}
```

Comment:
a way of exporting this stuff.  Should probably be a static method on TaskDefinition for better grouping.  Note that we always return copies so no one can muck with the defaults.